core-web/libs/edit-content/src/lib/fields/dot-edit-content-wysiwyg-field/dot-wysiwyg-plugin/dot-wysiwyg-plugin.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Editor } from 'tinymce';
+
+import { Injectable, NgZone, inject } from '@angular/core';
+
+import { DialogService } from 'primeng/dynamicdialog';
+
+import { filter, take } from 'rxjs/operators';
+
+import { DotUploadFileService } from '@dotcms/data-access';
+import { DotCMSContentlet } from '@dotcms/dotcms-models';
+import { DotAssetSearchDialogComponent } from '@dotcms/ui';
+
+import { formatDotImageNode } from './utils/editor.utils';
+
+@Injectable()
+export class DotWysiwygPluginService {
+    private readonly dialogService: DialogService = inject(DialogService);
+    private readonly dotUploadFileService: DotUploadFileService = inject(DotUploadFileService);
+    private readonly ngZone: NgZone = inject(NgZone);
+
+    initializePlugins(editor: Editor): void {
+        this.dotImagePlugin(editor);
+    }
+
+    private dotImagePlugin(editor: Editor): void {
+        editor.ui.registry.addButton('dotAddImage', {
+            icon: 'image',
+            onAction: () => {
+                this.ngZone.run(() => {
+                    const ref = this.dialogService.open(DotAssetSearchDialogComponent, {
+                        header: 'Insert Image',
+                        width: '800px',
+                        height: '500px',
+                        contentStyle: { padding: 0 },
+                        data: {
+                            assetType: 'image'
+                        }
+                    });
+
+                    ref.onClose
+                        .pipe(
+                            take(1),
```

Comment:
isn't the ref destroyed when you close the dialog?

Reply:
Right!